src/components/login/Login.tsx

```diff
@@ -0,0 +1,108 @@
+import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, TextField } from '@material-ui/core';
+import * as React from 'react';
+import * as rest from "../../rest/interface";
+import '../../styles.css';
+
+interface State {
+  open : boolean;
+  submitted : boolean
+  username : string
+  password : string
+}
+
+class Props {
+  public client: rest.Evergreen;
+}
+
+export class Login extends React.Component<Props, State> {
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      open : false,
+      submitted : false,
+      username : "",
+      password : "",
+    };
+  }
+
+  public render() {
+    return (
+      <div>
+        <Button onClick={this.handleClickOpen} color="inherit" className="login-button">
+          Log In
+        </Button>
+        <Dialog open={this.state.open} onClose={this.handleClickClose}>
+          <DialogTitle id="form-dialog-title">Log In</DialogTitle>
+          <DialogContent>
+            <DialogContentText>
+              Please enter your username and password. 
+            </DialogContentText>
+            <TextField
+              autoFocus={false}
+              margin="dense"
+              id="username"
+              label="username"
+              type="string"
+              fullWidth={true}
+              required={true}
+              onChange={this.updateSingleField("username")}
+              value={this.state.username}
+            />
+            <TextField
+              autoFocus={false}
+              margin="dense"
+              id="password"
+              label="password"
+              type="string"
+              fullWidth={true}
+              required={true}
+              onChange={this.updateSingleField("password")}
+              value={this.state.password}
+            />
+          </DialogContent>
+          <DialogActions>
+            <Button onClick={this.handleClickClose} color="primary">
+              Cancel
+            </Button>
+            <Button onClick={this.handleClickSubmit} color="primary">
+              Submit
+            </Button>
+          </DialogActions>
+        </Dialog>
+      </div>
+    );
+  }
+
+  private handleClickOpen = () => {
+    this.setState({
+      open : true,
+    });
+  }
+
+  private handleClickClose = () => {
+    this.setState({
+      open : false,
+      username: "",
+      password: "",
+    });
+  }
+
+  private handleClickSubmit = () => {
+    this.setState({
+      open : false,
+      submitted : true,
+    });
+  }
+ 
+  private updateSingleField = (fieldName:string) => {
```

Comment:
I didn't know about this before, but we can enforce that the field names passed into here have to be valid properties on the state object by using the "keyof" keyword in the function signature here. Could you try changing this function to only allow valid property names?